packages/@sanity/reflector-client/src/Connector.js

```diff
@@ -0,0 +1,89 @@
+/* global window */
+
+import getIt from 'get-it'
+import getItJsonResponse from 'get-it/lib/middleware/jsonResponse'
+import getItPromise from 'get-it/lib/middleware/promise'
+import Observable from '@sanity/observable/minimal'
+
+const EventSource = typeof window !== 'undefined' && window.EventSource
+? window.EventSource // Native browser EventSource
+: require('@sanity/eventsource') // Node.js, IE etc
+
+function parseEvent(event) {
+  try {
+    return (event.data && JSON.parse(event.data)) || {}
+  } catch (err) {
+    return err
+  }
+}
+export default class Connector {
+  constructor(sanityClient) {
+    this.sanityClient = sanityClient
+    this.request = getIt([
+      getItJsonResponse(),
+      getItPromise()
+    ])
+  }
+
+  listen(channel) {
+    const {token, withCredentials} = this.sanityClient.clientConfig
+
+    const esOptions = {}
+    if (token || withCredentials) {
+      esOptions.withCredentials = true
+    }
+
+    if (token) {
+      esOptions.headers = {
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    const url = this.sanityClient.getUrl(`presence/listen/${channel}`)
+    return new Observable(observer => {
+      const es = new EventSource(url, esOptions)
+      es.addEventListener('message', onMessage, false)
+
+      function onMessage(evt) {
+        const event = parseEvent(evt)
+        return event instanceof Error
+          ? observer.error(event)
+          : observer.next(event)
+      }
+
+      function unsubscribe() {
+        es.removeEventListener('message', onMessage, false)
+        es.close()
+      }
+
+      return unsubscribe
+    })
+  }
+
+  send(channel, message) {
+    const url = this.sanityClient.getUrl(`presence/send/${channel}`)
+
+    let headers = {}
+    const {token} = this.sanityClient.clientConfig
+    if (token) {
+      headers = {
+        Authorization: `Bearer ${token}`
+      }
+    }
+
+    return this.request({
```

Comment:
Any reason this returns a promise and not observable?

The case for making it return an observable is 1) it can be cancelled (e.g. if a component unmounts when a `send()` is in flight) and 2) it is trivial to convert an observable to a promise, but not the other way around.